Panic when try to list persistent volumes

**What steps did you take and what happened:**
Start octant. Click on Storage -> overview or Storage -> Persistent Volumes
After it octant crushes.

**What did you expect to happen:**
See a list of PV I have in cluster
**Anything else you would like to add:**
Logs from octant

```
I0424 22:11:10.716884   76620 request.go:947] Request Body: {"kind":"SelfSubjectAccessReview","apiVersion":"authorization.k8s.io/v1","metadata":{"creationTimestamp":null},"spec":{"resourceAttributes":{"namespace":"solo","verb":"get","resource":"persistentvolumeclaims"}},"status":{"allowed":false}}
I0424 22:11:10.716924   76620 round_trippers.go:419] curl -k -v -XPOST  -H "Accept: application/json, */*" -H "Content-Type: application/json" -H "User-Agent: octant/0.12.1" 'https://172.31.75.161:6443/apis/authorization.k8s.io/v1/selfsubjectaccessreviews'
I0424 22:11:10.722746   76620 round_trippers.go:438] POST https://172.31.75.161:6443/apis/authorization.k8s.io/v1/selfsubjectaccessreviews 201 Created in 5 milliseconds
I0424 22:11:10.722761   76620 round_trippers.go:444] Response Headers:
I0424 22:11:10.722767   76620 round_trippers.go:447]     Content-Type: application/json
I0424 22:11:10.722772   76620 round_trippers.go:447]     Content-Length: 238
I0424 22:11:10.722776   76620 round_trippers.go:447]     Date: Fri, 24 Apr 2020 18:11:10 GMT
I0424 22:11:10.722779   76620 round_trippers.go:447]     Cache-Control: no-cache, private
I0424 22:11:10.723085   76620 request.go:947] Response Body: {"kind":"SelfSubjectAccessReview","apiVersion":"authorization.k8s.io/v1","metadata":{"creationTimestamp":null},"spec":{"resourceAttributes":{"namespace":"solo","verb":"get","resource":"persistentvolumeclaims"}},"status":{"allowed":true}}
I0424 22:11:10.723306   76620 reflector.go:122] Starting reflector *unstructured.Unstructured (3m0s) from github.com/vmware-tanzu/octant/internal/objectstore/informer.go:82
I0424 22:11:10.723319   76620 reflector.go:160] Listing and watching *unstructured.Unstructured from github.com/vmware-tanzu/octant/internal/objectstore/informer.go:82
I0424 22:11:10.723391   76620 round_trippers.go:419] curl -k -v -XGET  -H "Accept: application/json" -H "User-Agent: octant/0.12.1" 'https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims?limit=500&resourceVersion=0'
I0424 22:11:10.723868   76620 round_trippers.go:419] curl -k -v -XGET  -H "Accept: application/json" -H "User-Agent: octant/0.12.1" 'https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-0'
I0424 22:11:10.728395   76620 round_trippers.go:438] GET https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims?limit=500&resourceVersion=0 200 OK in 4 milliseconds
I0424 22:11:10.728406   76620 round_trippers.go:444] Response Headers:
I0424 22:11:10.728410   76620 round_trippers.go:447]     Cache-Control: no-cache, private
I0424 22:11:10.728413   76620 round_trippers.go:447]     Content-Type: application/json
I0424 22:11:10.728416   76620 round_trippers.go:447]     Content-Length: 2553
I0424 22:11:10.728418   76620 round_trippers.go:447]     Date: Fri, 24 Apr 2020 18:11:10 GMT
I0424 22:11:10.729411   76620 request.go:947] Response Body: {"kind":"PersistentVolumeClaimList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/namespaces/solo/persistentvolumeclaims","resourceVersion":"4372492"},"items":[{"metadata":{"name":"data-solo-dev-consul-server-0","namespace":"solo","selfLink":"/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-0","uid":"1cbb0cc3-f22d-4bcd-b294-66d8f7857f82","resourceVersion":"3083691","creationTimestamp":"2020-04-08T09:47:42Z","labels":{"app":"consul","chart":"consul-helm","component":"server","hasDNS":"true","release":"dev"},"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"},"finalizers":["kubernetes.io/pvc-protection"]},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"local-pv-291b6f83","storageClassName":"local-storage","volumeMode":"Filesystem"},"status":{"phase":"Bound","accessModes":["ReadWriteOnce"],"capacity":{"storage":"1951Mi"}}},{"metadata":{"name":"data-solo-dev-consul-server-1","namespace":"solo","selfLink":"/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-1","uid":"a3b98465-7d72-4897-a700-53a473052abd","resourceVersion":"3083697","creationTimestamp":"2020-04-08T09:47:42Z","labels":{"app":"consul","chart":"consul-helm","component":"server","hasDNS":"true","release":"dev"},"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"},"finalizers":["kubernetes.io/pvc-protection"]},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"local-pv-d9962ba","storageClassName":"local-storage","volumeMode":"Filesystem"},"status":{"phase":"Bound","accessModes":["ReadWriteOnce"],"capacity":{"storage":"1951Mi"}}},{"metadata":{"name":"data-solo-dev-consul-server-2","namespace":"solo","selfLink":"/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-2","uid":"2bec196c-5b0a-405f-aa4c-8118a90072df","resourceVersion":"3083694","creationTimestamp":"2020-04-08T09:47:42Z","labels":{"app":"consul","chart":"consul-helm","component":"server","hasDNS":"true","release":"dev"},"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"},"finalizers":["kubernetes.io/pvc-protection"]},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"local-pv-ee63278d","storageClassName":"local-storage","volumeMode":"Filesystem"},"status":{"phase":"Bound","accessModes":["ReadWriteOnce"],"capacity":{"storage":"1951Mi"}}}]}
I0424 22:11:10.729846   76620 round_trippers.go:419] curl -k -v -XGET  -H "User-Agent: octant/0.12.1" -H "Accept: application/json" 'https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims?resourceVersion=4372492&timeoutSeconds=344&watch=true'
I0424 22:11:10.732808   76620 round_trippers.go:438] GET https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-0 200 OK in 8 milliseconds
I0424 22:11:10.732823   76620 round_trippers.go:444] Response Headers:
I0424 22:11:10.732829   76620 round_trippers.go:447]     Cache-Control: no-cache, private
I0424 22:11:10.732835   76620 round_trippers.go:447]     Content-Type: application/json
I0424 22:11:10.732839   76620 round_trippers.go:447]     Content-Length: 845
I0424 22:11:10.732844   76620 round_trippers.go:447]     Date: Fri, 24 Apr 2020 18:11:10 GMT
I0424 22:11:10.733865   76620 request.go:947] Response Body: {"kind":"PersistentVolumeClaim","apiVersion":"v1","metadata":{"name":"data-solo-dev-consul-server-0","namespace":"solo","selfLink":"/api/v1/namespaces/solo/persistentvolumeclaims/data-solo-dev-consul-server-0","uid":"1cbb0cc3-f22d-4bcd-b294-66d8f7857f82","resourceVersion":"3083691","creationTimestamp":"2020-04-08T09:47:42Z","labels":{"app":"consul","chart":"consul-helm","component":"server","hasDNS":"true","release":"dev"},"annotations":{"pv.kubernetes.io/bind-completed":"yes","pv.kubernetes.io/bound-by-controller":"yes"},"finalizers":["kubernetes.io/pvc-protection"]},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}},"volumeName":"local-pv-291b6f83","storageClassName":"local-storage","volumeMode":"Filesystem"},"status":{"phase":"Bound","accessModes":["ReadWriteOnce"],"capacity":{"storage":"1951Mi"}}}
I0424 22:11:10.735049   76620 round_trippers.go:438] GET https://172.31.75.161:6443/api/v1/namespaces/solo/persistentvolumeclaims?resourceVersion=4372492&timeoutSeconds=344&watch=true 200 OK in 5 milliseconds
I0424 22:11:10.735062   76620 round_trippers.go:444] Response Headers:
I0424 22:11:10.735066   76620 round_trippers.go:447]     Cache-Control: no-cache, private
I0424 22:11:10.735070   76620 round_trippers.go:447]     Content-Type: application/json
I0424 22:11:10.735073   76620 round_trippers.go:447]     Date: Fri, 24 Apr 2020 18:11:10 GMT
2020-04-24T22:11:10.746+0400	DEBUG	api/content_manager.go:151	generating content	{"client-id": "f354e02c-8656-11ea-93ff-acde48001122", "contentPath": "cluster-overview/storage/persistent-volumes", "elapsed": "37.009719ms"}
2020-04-24T22:11:10.746+0400	DEBUG	api/content_manager.go:97	stopping content manager	{"component": "websocket-client", "client-id": "f354e02c-8656-11ea-93ff-acde48001122"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x213766e]

goroutine 293 [running]:
github.com/vmware-tanzu/octant/internal/printer.getBoundPersistentVolumeClaim(0x38badc0, 0xc002b861c0, 0xc0009bc500, 0x0, 0x38dae40, 0xc0011cc9c0, 0x38bb180, 0xc0004280a8, 0x1, 0x1, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:217 +0xbe
github.com/vmware-tanzu/octant/internal/printer.createBoundPersistentVolumeClaimLink(0x38badc0, 0xc002b861c0, 0xc0009bc500, 0x0, 0x38dae40, 0xc0011cc9c0, 0x38bb180, 0xc0004280a8, 0xc00040f6e0, 0x0, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:241 +0x81
github.com/vmware-tanzu/octant/internal/printer.PersistentVolumeListHandler(0x38badc0, 0xc002b861c0, 0xc001364770, 0x0, 0x38dae40, 0xc0011cc9c0, 0x38bb180, 0xc0004280a8, 0x0, 0x0, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:41 +0x6c4
reflect.Value.call(0x23844c0, 0x36aa608, 0x13, 0x25a1949, 0x4, 0xc00014b128, 0x3, 0x3, 0x100beaf, 0x243afc0, ...)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x23844c0, 0x36aa608, 0x13, 0xc00014b128, 0x3, 0x3, 0xc00014b001, 0xc001ee2a40, 0xc00014b050)
	/usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:321 +0xb4
github.com/vmware-tanzu/octant/internal/printer.(*Resource).Print(0xc000a3b300, 0x38badc0, 0xc002b861c0, 0x388b780, 0xc001364770, 0x38bb1c0, 0xc0007fc630, 0xc001ee2a40, 0x97, 0x0, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/printer/printer.go:77 +0x472
github.com/vmware-tanzu/octant/internal/describer.(*List).Describe(0xc0013735f0, 0x38badc0, 0xc002b861c0, 0x0, 0x0, 0x38dae40, 0xc0011cc9c0, 0x38d5f40, 0xc001dc9780, 0xc000a62a80, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/describer/list.go:111 +0x919
github.com/vmware-tanzu/octant/internal/modules/clusteroverview.(*ClusterOverview).Content(0xc00136f180, 0x38badc0, 0xc002b861c0, 0xc0023a6190, 0x1b, 0xc00057e450, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/modules/clusteroverview/clusteroverview.go:167 +0x750
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).generateContent(0xc000e5c870, 0x38badc0, 0xc002b861c0, 0x38d9ae0, 0xc0003b4160, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:162 +0x321
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).runUpdate.func1(0x38badc0, 0xc002b861c0, 0x4578820)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:116 +0xe0
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run.func1(0x38badc0, 0xc000948480, 0x38cb740, 0xc00057e050, 0xc002e28e77, 0xc0013e08a0, 0xc000476190, 0xc000178270, 0x3b9aca00)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/poller.go:86 +0x271
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run(0xc000c16c20, 0x38badc0, 0xc000948480, 0xc0013e08a0, 0xc000178270, 0x3b9aca00)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/poller.go:95 +0x2ff
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).Start(0xc000e5c870, 0x38badc0, 0xc000948480, 0x38d9ae0, 0xc0003b4160, 0x3889080, 0xc00018cb80)
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:106 +0x1e3
created by github.com/vmware-tanzu/octant/internal/api.(*WebsocketState).Start
	/private/tmp/octant-20200424-56335-4w1et/src/github.com/vmware-tanzu/octant/internal/api/websocket_state.go:143 +0x9b
```
Kubectl get pv

```
NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                                STORAGECLASS    REASON   AGE
local-pv-291b6f83   1951Mi     RWO            Delete           Bound       solo/data-solo-dev-consul-server-0   local-storage            16d
local-pv-4ea947ca   1951Mi     RWO            Delete           Available                                        local-storage            16d
local-pv-93e1c02c   1951Mi     RWO            Delete           Available                                        local-storage            16d
local-pv-ca4da053   1951Mi     RWO            Delete           Available                                        local-storage            16d
local-pv-d9962ba    1951Mi     RWO            Delete           Bound       solo/data-solo-dev-consul-server-1   local-storage            16d
local-pv-ee63278d   1951Mi     RWO            Delete           Bound       solo/data-solo-dev-consul-server-2   local-storage            16d
```
Storage deployed with local volume provisioner.
```
apiVersion: v1
appVersion: 2.3.4
description: local provisioner chart
keywords:
- storage
- local
name: provisioner
version: 3.0.0
```
**Environment:**
octant version
```
Version:  0.12.1
Git commit:  870ee5775b84981ecc8f01e9a90f5182972c56bf
Built:  2020-04-24T10:29:28Z
```
kubectl version
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T20:52:22Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```
OS
`macOS 10.15.4`

--- a/changelogs/unreleased/889-GuessWhoSamFoo
+++ b/changelogs/unreleased/889-GuessWhoSamFoo
@@ -1,0 +1,1 @@
+Fixed case where unclaimed persistent volumes causes panic

--- a/internal/printer/persistentvolume.go
+++ b/internal/printer/persistentvolume.go
@@ -213,6 +213,10 @@ func getBoundPersistentVolumeClaim(ctx context.Context, pv *corev1.PersistentVol
 	pvc := &corev1.PersistentVolumeClaim{}
 
 	cr := pv.Spec.ClaimRef
+	if cr == nil {
+		return nil, nil
+	}
+
 	key := store.Key{
 		APIVersion: cr.APIVersion,
 		Kind:       cr.Kind,
@@ -244,6 +248,9 @@ func createBoundPersistentVolumeClaimLink(ctx context.Context, pv *corev1.Persis
 	}
 
 	cr := pv.Spec.ClaimRef
+	if cr == nil {
+		return component.NewLink("", "", ""), nil
+	}
 	claimText := fmt.Sprintf("%s/%s", cr.Namespace, cr.Name)
 	claimLink, err := options.Link.ForObject(pvc, claimText)
 	if err != nil {

--- a/internal/printer/persistentvolume_test.go
+++ b/internal/printer/persistentvolume_test.go
@@ -37,6 +37,12 @@ func Test_PersistentVolumeListHandler(t *testing.T) {
 		Items: []corev1.PersistentVolume{*object},
 	}
 
+	unbound := testutil.CreatePersistentVolume("unboundPersistentVolume")
+	unbound.CreationTimestamp = metav1.Time{Time: now}
+	unboundList := &corev1.PersistentVolumeList{
+		Items: []corev1.PersistentVolume{*unbound},
+	}
+
 	cases := []struct {
 		name     string
 		list     *corev1.PersistentVolumeList
@@ -62,6 +68,24 @@ func Test_PersistentVolumeListHandler(t *testing.T) {
 				}),
 		},
 		{
+			name: "unclaimed",
+			list: unboundList,
+			expected: component.NewTableWithRows("Persistent Volumes", "We couldn't find any persistent volumes!", cols,
+				[]component.TableRow{
+					{
+						"Name":           component.NewLink("", "unboundPersistentVolume", "/unboundPersistentVolume"),
+						"Capacity":       component.NewText("0"),
+						"Access Modes":   component.NewText(""),
+						"Reclaim Policy": component.NewText(""),
+						"Status":         component.NewText("Bound"),
+						"Claim":          component.NewLink("", "", ""),
+						"Storage Class":  component.NewText(""),
+						"Reason":         component.NewText(""),
+						"Age":            component.NewTimestamp(now),
+					},
+				}),
+		},
+		{
 			name:  "list is nil",
 			list:  nil,
 			isErr: true,
@@ -79,13 +103,13 @@ func Test_PersistentVolumeListHandler(t *testing.T) {
 			ctx := context.Background()
 
 			if tc.list != nil {
-				tpo.PathForObject(&tc.list.Items[0], tc.list.Items[0].Name, "/persistentVolume")
+				tpo.PathForObject(&tc.list.Items[0], tc.list.Items[0].Name, "/"+tc.list.Items[0].Name)
 
 				pvcKey, err := store.KeyFromObject(pvcObject)
 				require.NoError(t, err)
 
 				tpo.objectStore.EXPECT().Get(ctx, pvcKey).
-					Return(testutil.ToUnstructured(t, pvcObject), nil)
+					Return(testutil.ToUnstructured(t, pvcObject), nil).AnyTimes()
 
 				tpo.PathForObject(pvcObject, pvcObject.Namespace+"/"+pvcObject.Name, "/pvc")
 			}

